emcc.py

```diff
@@ -1529,20 +1529,24 @@ def check(input_file):
     if shared.Settings.EMBIND:
       forced_stdlibs.append('libembind')
 
-    if not shared.Settings.MINIMAL_RUNTIME and not shared.Settings.STANDALONE_WASM:
-      # The normal JS runtime depends on malloc and free so always keep them alive.
-      # MINIMAL_RUNTIME avoids this dependency as does STANDALONE_WASM mode (since it has no
-      # JS runtime at all).
-      shared.Settings.EXPORTED_FUNCTIONS += ['_malloc', '_free']
-
     if shared.Settings.WASM_BACKEND:
-      shared.Settings.EXPORTED_FUNCTIONS += ['_stackSave', '_stackRestore', '_stackAlloc']
       # We need to preserve the __data_end symbol so that wasm-emscripten-finalize can determine
       # the STATIC_BUMP value.
       shared.Settings.EXPORTED_FUNCTIONS += ['___data_end']
-      if not shared.Settings.STANDALONE_WASM:
-        # in standalone mode, crt1 will call the constructors from inside the wasm
-        shared.Settings.EXPORTED_FUNCTIONS.append('___wasm_call_ctors')
+
+    # In STANDALONE_WASM we try to avoid exports the stack allocations functions
+    # unless we know we will need them (e.g. for invoke_xxx)
+    if shared.Settings.SUPPORT_LONGJMP or not shared.Settings.STANDALONE_WASM:
```

Comment:
adding longjmp here seems confusing. is it for invokes as the comment says? if so then exceptions are also relevant.

but i worry this is adding extra complexity and risk. these functions should be removed by metadce anyhow in release builds?